### PR TITLE
Fixed activation key CLI test formatting

### PR DESCRIPTION
**Pull request by JacobCallahan**
A few other fixes and changes thrown in, but functionality
remains the same. Ran test suite with same results.

### CODE REVIEW

**Comment by JacobCallahan** [comment]
Have to make flake happy. wait for resubmission

File: tests/foreman/cli/test_activationkey.py [hunk 1]
```diff
@@ -58,173 +56,167 @@ def setUp(self):
...
-                'id': ackey['id']
-            }
-        )
+        result = ActivationKey.info({'id': ackey['id']})
 
         self.assertEqual(
             result.return_code,
             0,
-            "Activation key was not found: %s" % str(result.stderr))
+            'Activation key was not found: {}'.format(result.stderr))
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_activationkey.py]: Are we still doing `'{0}'` formatting for the sake of (better) Python 2.6 compatibility?

File: tests/foreman/cli/test_activationkey.py [hunk 2]
```diff
@@ -58,173 +56,167 @@ def setUp(self):
...
+        new_ackey_name = self._make_activation_key({u'name': name})['name']
         # Name should match passed data
         self.assertEqual(
-            new_ackey['name'],
-            test_data['name'],
-            ("Names don't match: '%s' != '%s'" %
-             (new_ackey['name'], test_data['name']))
+            new_ackey_name,
+            name,
+            'Names don\'t match: "{}" != "{}"'.format(new_ackey_name, name)
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_activationkey.py]: This is definitely a formatting improvement. Can I request that the error message be dropped completely? The redundant error message is redundant.

Comment (Ichimonji10) [inline tests/foreman/cli/test_activationkey.py]: Eck. I notice the same thing in many of these tests. Well, I'm happy to have a formatting improvement at least, and maybe it's best to revisit the topic of error messages at some other point in time.

File: tests/foreman/cli/test_activationkey.py [hunk 3]
```diff
@@ -437,21 +438,21 @@ def test_positive_delete_activation_key_1(self, test_data):
...
         except CLIFactoryError as err:
             self.fail(err)
 
-        result = ActivationKey.delete({'id': activation_key['id']})
+        result = ActivationKey.delete({'id': activation_key_id})
         self.assertEqual(
             result.return_code, 0, 'Failed to delete activation key')
         self.assertEqual(
-            len(result.stderr), 0, 'There should not be an error here')
+            len(result.stderr), 0, result.stderr)
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_activationkey.py]: So, so, *so* much better.

File: tests/foreman/cli/test_activationkey.py (hunk 1 above)
Comment (JacobCallahan) [inline tests/foreman/cli/test_activationkey.py]: i wasn't sure if it was this way or the other. i can put the 0 in though. i think that was the last conclusion

File: tests/foreman/cli/test_activationkey.py (hunk 2 above)
Comment (JacobCallahan) [inline tests/foreman/cli/test_activationkey.py]: i was iffy on removing these completely. i am willing to make the changes, though.

File: tests/foreman/cli/test_activationkey.py (hunk 1 above)
Comment (Ichimonji10) [inline tests/foreman/cli/test_activationkey.py]: Yeah. I'm not 100% sure if we're doing this, but I figure it's easy enough to add in the "0" that we might as well.

File: tests/foreman/cli/test_activationkey.py (hunk 2 above)
Comment (Ichimonji10) [inline tests/foreman/cli/test_activationkey.py]: Is the idea that both items being compared are shown, so given that no information is lost, we might as well add an error message? I think that's the idea. This is definitely a bit of a personal preference thing. I like having more terse code, given that QE folks will typically have to look at code when something fails anyway, so we might as well make the code pretty. But I'm not super set on that opinion, and can certainly live with the error message.

File: tests/foreman/cli/test_activationkey.py [hunk 4]
```diff
@@ -58,173 +56,167 @@ def setUp(self):
...
         # Use default organization if None are provided
-        if (
-                not options.get('organization', None)
-                and not options.get('organization-label', None)
-                and not options.get('organization-id', None)):
-            options['organization-id'] = self.org['id']
+        if (not options.get('organization', None) and
+                not options.get('organization-label', None) and
+                not options.get('organization-id', None)):
+                    options['organization-id'] = self.org['id']
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_activationkey.py]: Can you un-indent this line by eight spaces?

File: tests/foreman/cli/test_activationkey.py [hunk 5]
```diff
@@ -58,173 +56,167 @@ def setUp(self):
...
 
         """
-
-        new_ackey = self._make_activation_key(
-            {u'name': test_data['name'],
-             u'lifecycle-environment-id': self.env1['id']})
+        new_ackey_env = self._make_activation_key(
+            {u'name': name,
+             u'lifecycle-environment-id': self.env1['id']}
+        )['lifecycle-environment']
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_activationkey.py]: _make_activation_key({
        dict
        data
    })

File: tests/foreman/cli/test_activationkey.py (hunk 4 above)
Comment (sthirugn) [inline tests/foreman/cli/test_activationkey.py]: +1

**Comment by Ichimonji10** [comment]
ACK pending comments

**Comment by sthirugn** [comment]
ACK pending open comments. Good work!

File: tests/foreman/cli/test_activationkey.py (hunk 5 above)
Comment (elyezer) [inline tests/foreman/cli/test_activationkey.py]: :+1:

File: tests/foreman/cli/test_activationkey.py (hunk 4 above)
Comment (JacobCallahan) [inline tests/foreman/cli/test_activationkey.py]: yep. got carried away!

**Comment by JacobCallahan** [comment]
All comments addressed

**Pull request merged**